src/components/EmojiPicker/EmojiPickerMenu/index.native.js

```diff
@@ -1,76 +1,41 @@
 import useSingleExecution from '@hooks/useSingleExecution';
 import useStyleUtils from '@hooks/useStyleUtils';
 import useThemeStyles from '@hooks/useThemeStyles';
 import useWindowDimensions from '@hooks/useWindowDimensions';
-import compose from '@libs/compose';
-import * as EmojiUtils from '@libs/EmojiUtils';
-import * as User from '@userActions/User';
 import CONST from '@src/CONST';
-import ONYXKEYS from '@src/ONYXKEYS';
+import BaseEmojiPickerMenu from './BaseEmojiPickerMenu';
+import {emojiPickerMenuDefaultProps, emojiPickerMenuPropTypes} from './emojiPickerMenuPropTypes';
+import useEmojiPickerMenu from './useEmojiPickerMenu';
 
-const propTypes = {
-    /** Function to add the selected emoji to the main compose text input */
-    onEmojiSelected: PropTypes.func.isRequired,
+const propTypes = emojiPickerMenuPropTypes;
 
-    /** Stores user's preferred skin tone */
-    preferredSkinTone: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+const defaultProps = emojiPickerMenuDefaultProps;
 
-    /** Stores user's frequently used emojis */
-    // eslint-disable-next-line react/forbid-prop-types
-    frequentlyUsedEmojis: PropTypes.arrayOf(PropTypes.object),
-
-    /** Props related to translation */
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
-    frequentlyUsedEmojis: [],
-};
-
-function EmojiPickerMenu({preferredLocale, onEmojiSelected, preferredSkinTone, translate, frequentlyUsedEmojis}) {
+function EmojiPickerMenu({onEmojiSelected}) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
-    const emojiList = useAnimatedRef();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const allEmojis = useMemo(() => EmojiUtils.mergeEmojisWithFrequentlyUsedEmojis(emojis), [frequentlyUsedEmojis]);
-    const headerEmojis = useMemo(() => EmojiUtils.getHeaderEmojis(allEmojis), [allEmojis]);
-    const headerRowIndices = useMemo(() => _.map(headerEmojis, (headerEmoji) => Math.floor(headerEmoji.index / CONST.EMOJI_NUM_PER_ROW)), [headerEmojis]);
-    const [filteredEmojis, setFilteredEmojis] = useState(allEmojis);
-    const [headerIndices, setHeaderIndices] = useState(headerRowIndices);
     const {windowWidth} = useWindowDimensions();
+    const {translate} = useLocalize();
+    const {allEmojis, headerEmojis, headerRowIndices, filteredEmojis, headerIndices, setFilteredEmojis, setHeaderIndices, isListFiltered, suggestEmojis, preferredSkinTone} =
+        useEmojiPickerMenu();
     const {singleExecution} = useSingleExecution();
-
-    useEffect(() => {
-        setFilteredEmojis(allEmojis);
-    }, [allEmojis]);
-
-    useEffect(() => {
-        setHeaderIndices(headerRowIndices);
-    }, [headerRowIndices]);
-
-    const getItemLayout = (data, index) => ({length: CONST.EMOJI_PICKER_ITEM_HEIGHT, offset: CONST.EMOJI_PICKER_ITEM_HEIGHT * index, index});
+    const emojiList = useAnimatedRef();
```

Comment:
I think we _might_ be able to use `useAnimatedRef` and `scrollTo` on web as well, so that could be another difference between these files that could be eliminated.

When we follow-up to add `ArrowRight` and `ArrowLeft` handling to `useArrowKeyFocusManager`, then the ref and imperative scrolling logic can be moved into `BaseEmojiPickerMenu`.